cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -505,14 +505,16 @@ bool do_iterate_type_impl(
 }
 
 template<class KeyBucketizer>
-ObjectSizes do_calculate_sizes_for_type_impl(
+void do_visit_object_sizes_for_type_impl(
     KeyType key_type,
     const std::string& root_folder,
     const std::string& bucket_name,
     const S3ClientInterface& s3_client,
     KeyBucketizer&& bucketizer,
-    const PrefixHandler& prefix_handler = default_prefix_handler(),
-    const std::string& prefix = std::string{}) {
+    const PrefixHandler& prefix_handler,
+    const std::string& prefix,
+    const ObjectSizesVisitor& visitor
+    ) {
     ARCTICDB_SAMPLE(S3StorageCalculateSizesForType, 0)
 
     auto path_info = calculate_path_info(root_folder, key_type, prefix_handler, prefix, std::move(bucketizer));
```

Comment:
I know it's not part of the PR but it would be technically more correct to std::forward the bucketizer as it's a forwarding reference.